# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/tcp_wrappers-10-uclibc.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -u tcp_wrappers_7.6/Makefile tcp_wrappers_7.6/Makefile
--- tcp_wrappers_7.6/percent_m.c.orig	2004-11-23 10:11:29.789173760 +0100
+++ tcp_wrappers_7.6/percent_m.c	2004-11-23 10:31:16.635745656 +0100
@@ -29,11 +25,15 @@
 
     while (*bp = *cp)
 	if (*cp == '%' && cp[1] == 'm') {
+#if 0
 	    if (errno < sys_nerr && errno > 0) {
 		strcpy(bp, sys_errlist[errno]);
 	    } else {
 		sprintf(bp, "Unknown error %d", errno);
 	    }
+#else
+	    strcpy(bp, (strerror(errno)? strerror(errno) : "Unknown error"));
+#endif
 	    bp += strlen(bp);
 	    cp += 2;
 	} else {
